File: app/src/main/java/be/vergauwen/simon/cleanmvp/ui/TestPresenter.java
package be.vergauwen.simon.cleanmvp.ui;

import android.util.Log;
import be.vergauwen.simon.cleanmvp.mvp.MVPPresenter;

public class TestPresenter extends MVPPresenter<TestContract.View>
    implements TestContract.Presenter {

    @Override
    public void attachView(TestContract.View view) {
        super.attachView(view);
    }

    @Override
    public void loadThings() {
        Log.e("TestPresenter",".loadThings()");
        getView().showThings();
    }
}

File: app/src/main/kotlin/be/vergauwen/simon/cleanmvp/ui/TestPresenter.kt
package be.vergauwen.simon.cleanmvp.ui

import android.util.Log
import be.vergauwen.simon.cleanmvp.mvp.MVPPresenter

class TestPresenter : MVPPresenter<TestContract.View>(), TestContract.Presenter {

  override fun loadThings() {
    Log.e("TestPresenter", ".loadThings()")
    getView()?.showThings()
  }
}